src/sfizz/FilePool.cpp

```diff
@@ -587,17 +587,24 @@ void sfz::FilePool::setRamLoading(bool loadInRam) noexcept
 
 void sfz::FilePool::triggerGarbageCollection() noexcept
 {
-    const std::unique_lock<SpinMutex> lastUsedLock { lastUsedMutex, std::try_to_lock };
-    const std::unique_lock<SpinMutex> garbageLock { garbageMutex, std::try_to_lock };
-    if (!lastUsedLock.owns_lock() || !garbageLock.owns_lock())
+    const std::unique_lock<SpinMutex> guard { garbageAndLastUsedMutex, std::try_to_lock };
+    if (!guard.owns_lock())
         return;
 
     const auto now = std::chrono::high_resolution_clock::now();
     swapAndPopAll(lastUsedFiles, [&](const FileId& id) {
         if (garbageToCollect.size() == garbageToCollect.capacity())
            return false;
 
-        auto& data = preloadedFiles[id];
+        auto it = preloadedFiles.find(id);
+        if (it == preloadedFiles.end()) {
+            // Getting here means that the preloadedFiles got changed (probably cleared)
+            // while the lastUsedFiles were untouched.
+            ASSERTFALSE;
```

Comment:
We should not keep this assert, given that this condition is possible, is that correct?

Reply:
Well the `if` handles the error gracefully if this happens in Release: the file id is assumed invalid and remove from the list of recent files. Assuming the clearing is done properly it should not happen though. I'd keep the assertion as a way to catch an unexpected code path signaling a deeper error.